tests/test_templatetags.py

```diff
@@ -761,9 +839,222 @@ def test_isolated_slot_is_error(self):
                 {% component_block "broken_component" %}
-                    {% slot "header" %}Custom header{% endslot %}
-                    {% slot "main" %}Custom main{% endslot %}
-                    {% slot "footer" %}Custom footer{% endslot %}
+                    {% fill "header" %}Custom header {% endfill %}
+                    {% fill "main" %}Custom main{% endfill %}
+                    {% fill "footer" %}Custom footer{% endfill %}
+                {% endcomponent_block %}
+            """
+            ).render(Context({}))
+
+    def test_non_unique_fill_names_is_error(self):
+        with self.assertRaises(TemplateSyntaxError):
+            Template(
+                """
+                {% load component_tags %}
+                {% component_block "broken_component" %}
+                    {% fill "header" %}Custom header {% endfill %}
+                    {% fill "header" %}Other header{% endfill %}
                 {% endcomponent_block %}
             """
             ).render(Context({}))
+
+    def test_non_unique_slot_names_is_error(self):
+        with self.assertRaises(TemplateSyntaxError):
+            Template(
+                """
+                {% load component_tags %}
+                {% component_block "nonunique_slot_component" %}
+                {% endcomponent_block %}
+                """
+            ).render(Context({}))
+
+
+class ComponentNestingTests(SimpleTestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        component.registry.register("dashboard", _DashboardComponent)
+        component.registry.register("calendar", _CalendarComponent)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        super().tearDownClass()
+        component.registry.clear()
+
+    def test_component_nesting_component_without_fill(self):
+        template = Template(
+            """
+            {% load component_tags %}
+            {% component "dashboard" %}
+            """
+        )
+        rendered = template.render(Context({"items": [1, 2, 3]}))
+        expected = """
+        <div class="dashboard-component">
+          <div class="calendar-component">
+            <h1>
+              Welcome to your dashboard!
+            </h1>
+            <main>
+              Here are your to-do items for today:
+            </main>
+          </div>
+          <ol>
+            <li>1</li>
+            <li>2</li>
+            <li>3</li>
+          </ol>
+        </div>
+        """
+        self.assertHTMLEqual(rendered, expected)
+
+    def test_component_nesting_component_with_fill_and_super(self):
+        template = Template(
+            """
+            {% load component_tags %}
+            {% component_block "dashboard" %}
+              {% fill "header" as "h" %} Hello! {{ h.default }} {% endfill %}
+            {% endcomponent_block %}
+            """
+        )
+        import sys
+
+        sys.setrecursionlimit(100)
```

Comment:
How long does this setting apply for? Should this be reset not to mess with the users settings? Maybe a try-finally block that resets?

Reply:
> How long does this setting apply for

Embarrassingly permanently. This... should never have made it into the commit. Thanks for catching it! 🙈